lib/logstash/codecs/multiline.rb

```diff
@@ -177,14 +183,19 @@ def flush(&block)
 
   def do_next(text, matched, &block)
     buffer(text)
-    flush(&block) if !matched
+    flush(&block) if !matched || maximum_buffered_lines?
   end
 
   def do_previous(text, matched, &block)
-    flush(&block) if !matched
+    flush(&block) if !matched || maximum_buffered_lines?
     buffer(text)
   end
 
+
+  def maximum_buffered_lines?
+    @buffer.size > @max_lines
```

Comment:
Also can we log that this capacity (size or bytes) was reached and we are truncating? this will bring users to pay attention ... hopefully